openshift/openshift_transport.go

```diff
@@ -139,6 +103,7 @@ func (ref openshiftReference) DockerReference() reference.Named {
 func (ref openshiftReference) PolicyConfigurationIdentity() string {
 	res, err := policyconfiguration.DockerReferenceIdentity(ref.dockerReference)
 	if res == "" || err != nil { // Coverage: Should never happen, NewReference constructs a valid tagged reference.
+		// TODO(runcom,mitr): it's awful to panic inside a library
```

Comment:
Can't this just return an error w/o totally failing NewReference (if we move this check there)?